keyboards/skyloong/dt40/keymaps/default/keymap.c

```diff
@@ -0,0 +1,62 @@
+/* Copyright 2021 linlin
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include QMK_KEYBOARD_H
+
+// Defines names for use in layer keycodes and the keymap
+enum layer_names {
+    _BASE,
+    _FN
+};
+
+// Defines the keycodes used by our macros in process_record_user
+enum custom_keycodes {
+    QMKBEST = SAFE_RANGE,
+    QMKURL
+};
+
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+    /* Base */
+    [_BASE] = LAYOUT(
+        KC_A,    KC_1,    MO(_FN),
+            KC_TAB,   KC_SPC
+    ),
+    [_FN] = LAYOUT(
+        QMKBEST, QMKURL,  _______,
```

Comment:
And you haven't read the [PR Checklist](https://docs.qmk.fm/#/pr_checklist).

Reply:
> And you haven't read the [PR Checklist](https://docs.qmk.fm/#/pr_checklist).

This looks like generated from the template, or intactly copied from the doc. Going through the whole PR, my guess is the default keymap doesn't matter, even the whole PR doesn't. It's just a placeholder here so that the author could submit his VIA json to the-via/keyboards repo (merged QMK PR is a prerequisite) 😆

Reply:
The author's website shows that they are a commercial team selling keyboards running QMK. I'm totally fine with it as long as the QMK team feels OK. However, they should at least ensure the code quality following the contribution conducts.

Reply:
This project is indeed transplanted and modified from other projects. Let's check it carefully again. I hope I can do something useful to enrich the layout of the keyboard.